### PR TITLE
Allow non-source directory builds and no-plugin builds

### PR DESCRIPTION
**Pull request by pothos**
Hello,

what do you think about that?
This enables to build with e.g. jhbuild/gnome-continuous. The autogen.sh comes from GNOME Disks and could be improved I guess(?). I also ran into a linking error if all plugins were disabled.

Best regards,
Kai

### CODE REVIEW

**Comment by vpodzime** [comment]
Jenkins, this is ok to test.

File: autogen.sh [hunk 1]
```diff
@@ -1,7 +1,33 @@
...
+        echo "*** WARNING: I am going to run \`configure' with no arguments." >&2
+        echo "*** If you wish to pass any to it, please specify them on the" >&2
+        echo "*** \`$0\' command line." >&2
+        echo "" >&2
+fi
+
+# libtoolize --copy --force --install
+aclocal --install || exit 1
+autoreconf --verbose --force --install -Wno-portability || exit 1
+cd $olddir
```
Comment (vpodzime) [inline autogen.sh]: Can't this use ``pushd`` and ``popd`` instead of ``cd`` and remembering the ``$olddir``?

File: autogen.sh [hunk 2]
```diff
@@ -1,7 +1,33 @@
...
+if [ "$NOCONFIGURE" = "" ]; then
```
Comment (vpodzime) [inline autogen.sh]: This shouldn't be the default.

**Comment by vpodzime** [comment]
@pothos, could you please update this so that it can be merged?

**Comment by pothos** [comment]
(Same additional jhbuild declaration needed as the respective PR in libbytesize)

File: autogen.sh [hunk 3]
```diff
@@ -1,7 +1,17 @@
...
-libtoolize --copy --force
-aclocal -I m4
-autoconf
-automake --foreign --add-missing --copy
-rm -rf autom4te.cache
+test -n "$srcdir" || srcdir=`dirname "$0"`
+test -n "$srcdir" || srcdir=.
+
+pushd .
+cd $srcdir
```
Comment (vpodzime) [inline autogen.sh]: This should be ``pushd $srcdir``, which does the same as the two lines above.

**Pull request merged**